### ISSUE TITLE
Screen locking service keeps getting disabled automatically

### ISSUE THREAD

**Issue by akhil-panicker** [8bitPit/Niagara-Issues#2099]
### Describe the bug

Screen locking service keeps getting disabled automatically.
I need to frequently go to accessibility settings to enable it, only for it to get disabled automatically.

### Steps to reproduce

1. Enable Niagara button
2. Press and hold the button 
3. Edit tap action 
4. Select Lock Screen option 
5. Apply the changes and close the settings 
6. The tap action works fine
7. After a few mins or 1-2 hours, the action gets disabled and i need to enable it from accessibility settings again.

### Does the issue also occur with other 3rd-party launchers?

Problem doesn't occur with other launchers

### Other 3rd-party launcher(s) tested

Used to happen in Nova launcher (the one I used before Niagara) but got resolved later.

### Device name

Vivo V20 Pro

### Android version

Android 12

### Screenshots / Screen recordings

![Screenshot_20221029_234640](https://user-images.githubusercontent.com/73038767/198847196-1e104d05-b66d-481e-bc27-a3657eba50ef.jpg)

**Comment by Just-SKY** [comment]
In order to stop this issue, you have to enable "autostart" permission to the app. It used to happened to me until I turn on that option.

**Comment by akhil-panicker** [comment]
> 

It was already enabled on my phone. PFB screenshot.
![Screenshot_20221101_132758](https://user-images.githubusercontent.com/73038767/199186185-11698730-251f-497e-9b15-3b2d68bf4132.jpg)

**Comment by akhil-panicker** [comment]
Played around the battery settings and I may have come across a solution.

The background power management setting was set to smart control (due to which the app was probably getting killed). I changed it to _don't restrict..._ and it is working fine now.🤞

Refer the below screenshot for the relevant setting.
![Screenshot_20221123_113534](https://user-images.githubusercontent.com/73038767/203482039-5cc61f32-32c7-48f0-9e95-33a9db0dcaed.jpg)

Closing the ticket for now. Hope this helps others facing the same issue.
